dotnet5 changes

Schlebbe opened the pull request.

Commit:
Moved Initialize configuration to extension method

Commit:
Fixed schedule job starting on startup

ovelartelius committed:
:memo: Updated description text in the job

ovelartelius committed:
:memo: Updated version to v0.3

ovelartelius committed:
Removed nuspec file

ovelartelius committed:
:art: removed jpg and ico from project

ovelartelius committed:
Added nupkg until feed is up and running.

ovelartelius committed:
Started to update readme documentation

ovelartelius committed:
Update README.md

ovelartelius committed:
Update README.md

ovelartelius committed:
Added startup example

vansterhant commented on README.md:
```diff
@@ -9,26 +9,66 @@ The synchronizer can be run as a InitializationModule or as a ScheduledJob. It d
+			_configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+			...
+		}
+		public void ConfigurationServices(IServiceCollection services)
```
Spelling mistake of the method It's called "ConfigureServices"

vansterhant commented on Addon.Episerver.EnvironmentSynchronizer/Configuration/ServiceCollectionExtensions.cs:
```diff
@@ -0,0 +1,19 @@
+    {
+        public static IServiceCollection AddEnvironmentSynchronization(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddSingleton<IConfigurationReader, ConfigurationReader>();
```
If the dependencies are added this way, one should drop the Serviceconfiguration-attributes, right? 
[ServiceConfiguration(typeof(IConfigurationReader))]

vansterhant reviewed (changes requested):
The I read the code the DI-registrations are now doubled and for the EnvironmentSynchronizationManager registered both on the interface and the implementation?

[ServiceConfiguration(ServiceType = typeof(IEnvironmentSynchronizationManager))]

Commit:
Fixed README spelling errors

Commit:
Removed ServiceConfiguration attribute

Commit:
Removed ServiceConfiguration attribute

ovelartelius committed:
Update EnvironmentSynchronizationManager.cs

Use environment variable ASPNETCORE_ENVIRONMENT as specified in https://docs.developers.optimizely.com/digital-experience-platform/v1.2.0-dxp-cloud-services/docs/environment-configurations

Schlebbe committed:
Merge branch 'master' into feature/dotnet5

ovelartelius committed:
Added yml to build nuget

ovelartelius committed:
Update to correct minor

ovelartelius committed:
Removed err value in yml

ovelartelius committed:
Start convert to Xunit tests

ovelartelius committed:
Removed MS test NuGets
